### PR TITLE
TC 74 Add card positioned at bottom always

### LINKED ISSUE #74: Make the add card fixed at bottom when scrolling through cards

### PR DESCRIPTION
**Pull request by rizwan171**
resolves #74 and #91

### CODE REVIEW

**Comment by rizwan171** [comment]
**Steps to test**
1. Pull, branch, load app, create a list
2. Add cards to the list such that you get a scroll bar
3. Confirm that when you scroll, the Add Card text is always at the bottom of the list
4. Scroll to the top of the list and click Add Card
5. Confirm that the list scrolls to the bottom so the text box and Add button are visible

File: src/components/List/AddCard/AddCardForm/AddCardForm.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef, useState } from "react";
+import { FiX } from "react-icons/fi";
+import { useDispatch } from "react-redux";
+import { addCard } from "../../../../features/cardsSlice";
+
+const AddCardForm = ({ setOpen, open, listId }) => {
+  const dispatch = useDispatch();
+  const [text, setText] = useState("");
+  const inputRef = useRef(null);
+  const inlineStyle = open ? { height: inputRef.current?.scrollHeight } : { height: 0, overflow: "hidden" };
```
Comment (rizwan171) [inline src/components/List/AddCard/AddCardForm/AddCardForm.js:10]: This inlineStyle controls the visibility of the element. If it isnt open, the height is set to 0

File: src/components/List/AddCard/AddCardPrompt/AddCardPrompt.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React, { useRef } from "react";
+import { FiPlus } from "react-icons/fi";
+
+const AddCardPrompt = ({ open, setOpen }) => {
+  const ref = useRef();
+  const inlineStyle = open ? { height: ref.current?.scrollHeight } : { height: 0, overflow: "hidden" };
```
Comment (rizwan171) [inline src/components/List/AddCard/AddCardPrompt/AddCardPrompt.js:6]: Like above, this inlineStyle controls the visibility of the element. If it isnt open, the height is set to 0

File: src/components/List/AddCard/AddCardPrompt/AddCardPrompt.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React, { useRef } from "react";
```
Comment (rizwan171) [inline src/components/List/AddCard/AddCardPrompt/AddCardPrompt.js:1]: This component shows the Add Card at the bottom of the list

File: src/components/List/AddCard/AddCardForm/AddCardForm.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useRef, useState } from "react";
```
Comment (rizwan171) [inline src/components/List/AddCard/AddCardForm/AddCardForm.js:1]: This component shows the text area and Add Card green button when you've pressed the Add Card at the bottom of the list

File: src/components/List/main/List.js [hunk 5]
```diff
@@ -22,17 +24,16 @@ const List = ({ list, index }) => {
...
-              </div>
+              <>
+                <div ref={provided.innerRef} {...provided.droppableProps} className="scroll-y-hidden overflow-x-hidden">
+                  {cards.map((card, index) => (
+                    <ListCard key={card.id} card={card} index={index} />
+                  ))}
+                  <AddCardForm listId={list.id} open={open} setOpen={setOpen} />
+                  {provided.placeholder}
+                </div>
+                <AddCardPrompt open={!open} setOpen={setOpen} />
```
Comment (rizwan171) [inline src/components/List/main/List.js:35]: Something to note here. AddCardForm and AddCardPrompt both share one state value `open` which controls their visibility.
Though they both take opposite values i.e. `open={open}` for AddCardForm, and `open={!open}` for AddCardPrompt.
This allows only one of them to ever be visible at once

**Review by nk711 — CHANGES_REQUESTED** [review]
Functionality wise the Add Card is positioned correctly now.

Main Issues:
1. Pressing the 'Add Card' button does not add a new card to the list.
2. Styling for the Add Card input seems to have been removed.

Whilst your doing styling;
3. Make sure the top and bottom padding of the 'Add a card' button are the same.

**Comment by rizwan171** [comment]
This change also seems to have regressed #34

**Pull request merged**